geospaas_harvesting/ingesters.py

```diff
@@ -539,15 +540,18 @@ def _get_normalized_attributes(self, dataset_info, *args, **kwargs):
         # set optional ForeignKey metadata from Nansat or from defaults
         normalized_attributes['gcmd_location'] = n_metadata.get(
             'gcmd_location', pti.get_gcmd_location('SEA SURFACE'))
-        normalized_attributes['provider'] = n_metadata.get(
-            'data_center', pti.get_gcmd_provider('NERSC'))
+        normalized_attributes['provider'] = pti.get_gcmd_provider(n_metadata.get('provider', 'NERSC'))
         normalized_attributes['iso_topic_category'] = n_metadata.get(
             'ISO_topic_category', pti.get_iso19115_topic_category('Oceans'))
 
         # Find coverage to set number of points in the geolocation
         if len(nansat_object.vrt.dataset.GetGCPs()) > 0:
             nansat_object.reproject_gcps()
         normalized_attributes['location_geometry'] = GEOSGeometry(
-            nansat_object.get_border_wkt(nPoints=n_points), srid=4326)
+            nansat_object.get_border_wkt(n_points=n_points), srid=4326)
+
+        if n_metadata.get('dataset_parameters', None):
+            normalized_attributes['dataset_parameters'] = [get_cf_or_wkv_standard_name(dataset_param)
+                for dataset_param in json.loads(n_metadata['dataset_parameters'])]
```

Comment:
- you look up the `'dataset_parameters'` key twice in the `n_metadata` dictionary. Just store the result in a variable the first time.
 - what happens if `n_metadata['dataset_parameters']` is not valid json? Or if it is not a list?
 - I find the way the list comprehension is indented quite confusing. Can you do something like the following, or use a regular loop?
     ```python
    new_list = [
        some_function(loop_var)
        for loop_var in some_list
    ]
    ```

Reply:
what do you mean by the third bullet? what you written is the same style of my code!

Reply:
In my example, the two parts of the list comprehension are at the same indentation level, making it easier to read.